### DISCUSSION TITLE
Route::compareWithParameterOnRouteName

### DISCUSSION THREAD

**Discussion by tetthys** [laravel/framework#51999] [Ideas]
```php

        $category = (object) ['id' => 1];
        
        Route::compareWithParameterOnRouteName(
            'category.index',
            'categoryId',
            function (mixed $categoryId) use ($category) {
                return $categoryId === $category->id;
            }
        );
```

It will be useful when highlighting a DOM element.

Especially in a blade component to which is passed an iterated model.

If `Route` shouldn't do this, `Request::compareWithParameterOnRouteName` might be good.

**Comment by ghost** [comment]
This seems to me like a specific case for your logic. I can't understand what you want to achieve with this...

**Reply by shaedrich** [reply]
Maybe, @tetthys can expand their example a little, so we get a better grasp of where it would be useful 🤔
